tests/kafkatest/tests/core/compatibility_test_new_broker_test.py

```diff
@@ -46,12 +45,12 @@ def setUp(self):
 
     @cluster(num_nodes=6)
     @parametrize(producer_version=str(LATEST_0_8_2), consumer_version=str(LATEST_0_8_2), compression_types=["none"], new_consumer=False, timestamp_type=None)
-    @parametrize(producer_version=str(LATEST_0_8_2), consumer_version=str(LATEST_0_9), compression_types=["none"], new_consumer=False, timestamp_type=None)
```

Comment:
I can't discern a pattern apart from "have some coverage for all the versions" at the moment. Starting from a clean slate, if we want to be completely distrustful of all parts of the system, we'd want to test every possible producer/consumer combination. But we test `same version` producer/consumer pairs in many other tests, so we could drop those perhaps. So, this test would be about testing producer/consumers with different versions. The numbers will grow pretty quickly if we don't drop support for older versions though.

What do you think?